usb.c

```diff
@@ -1120,8 +1121,12 @@ int usb_context_scan(struct iio_scan_backend_context *ctx,
 		unsigned int interface = 0;
 
 		ret = libusb_open(dev, &hdl);
-		if (ret)
+		if (ret) {
+			if (ret != LIBUSB_ERROR_ACCESS)
+				perms = true;
 			continue;
+		}
+		perms = true;
```

Comment:
Should probably only be set if the device is a IIO device. There might be other devices in the system that are accessible to the user. There might be other devices in the system that are accessible, but this does not mean that there is a IIO udev rule.

And in that case perms = false will be equivalent of simply no IIO device being found. So I think we can just skip the patch and simply print the warning when no IIO contexts have been found.